File: src/main/go/string/valid_palindrome.go
package string

import (
	"strings"
	"unicode"
)

func isPalindrome(s string) bool {

	s = strings.ToLower(s)

	left := 0
	right := len(s) - 1

	for left <= right {
		if !unicode.IsLetter(rune(s[left])) && !unicode.IsDigit(rune(s[left])) {
			left++
		} else if !unicode.IsLetter(rune(s[right])) && !unicode.IsDigit(rune(s[right])) {
			right--
		} else {
			if s[right] != s[left] {
				return false
			}
			left++
			right--
		}

	}

	return true
}

File: src/main/python/leetcode/string/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        left = 0
        right = len(s)-1

        while left <= right:
            if s[left].isdigit() == False and s[left].isalpha() == False:
                left = left+1
            elif s[right].isdigit() == False and s[right].isalpha() == False:
                right = right - 1
            else:
                if s[right].lower() != s[left].lower():
                    return False
                left = left + 1
                right = right - 1

        return True
